### DISCUSSION TITLE
How to properly implement a restart of multiple servers Unturned after updating the game?

### DISCUSSION THREAD

**Discussion by d3adwolf** [GameServerManagers/LinuxGSM#3893] [Q&A]
Greetings, everyone,

I have 9 servers on the game Unturned, and when I update the game through the command `./untserver update` through crontab, servers are not restarted, or the game is updated, and the servers continue to run on the old version, or the game is updated, but the servers are off and all. 

I made a script `.sh` to restart the servers, I thought to add it to some file in `/lgsm/functions/`, so that when the game was successfully updated running restart script servers, but when you update LGSM files and scripts overwritten, including those associated with updating the game.

So I would like to know how to properly implement this whole thing, so that when the game is successfully updated, restarted all 9 servers.

Thanks in advance for the tips and help.
<br>

 ### Tech. Details:
 - Ubuntu 22.04 LTS
 - LGSM 22.1.0
 - Unturned 3.22.11.3

**Comment by visdmin** [comment]
Hey @d3adwolf, have you found a solution to your problem yet ?
In case you haven't I could try to help you out tomorrow.

**Reply by visdmin** [reply]
For starters here is an example that could help you, script that i wrote sometime ago for very similar issue (someone here had similar issue with updating and restarting `Rust` servers).

In case this is something that you are looking for we have modify the script for your needs, and you can use it via crontab to keep your servers updated, and restarted when ever they are updated.

`rust_update_restart-sh`:
https://gist.github.com/visdmin/22916bae2500d052b0a55296b34dc1a3

**Reply by d3adwolf** [reply]
Hi, no, I put "untserver update" in the crontab every 3-5 minutes, and the only thing it does: it downloads the update to the game in time without restarting it. And made a separate `srv_restart.sh` to restart several servers at once. As for your script, I'm currently out of town, but in the next few days will look, I understand it will need to prescribe it in the crontab instead of `untserver`.

**Reply by d3adwolf** [reply]
> For starters here is an example that could help you, script that i wrote sometime ago for very similar issue (someone here had similar issue with updating and restarting `Rust` servers).
> 
> In case this is something that you are looking for we have modify the script for your needs, and you can use it via crontab to keep your servers updated, and restarted when ever they are updated.
> 
> `rust_update_restart-sh`: https://gist.github.com/visdmin/22916bae2500d052b0a55296b34dc1a3

Yes, I need help with that, especially your script can not remake Unturned, although I may not have enough experience in Bash.
